lib/matrix_convert_to_zero.rb

```diff
@@ -1,10 +1,25 @@
-# Updates the input matrix based on the following rules:
-# Assumption/ Given: All numbers in the matrix are 0s or 1s
-# If any number is found to be 0, the method updates all the numbers in the
-# corresponding row as well as the corresponding column to be 0.
+# Time complexity: O(n*m), where n and m are nums of rows and columns
+# Space complexity: O(n + m), where n and m are nums of rows and columns
 
-# Time complexity: ?
-# Space complexity: ?
 def matrix_convert_to_zero(matrix)
-  raise NotImplementedError
-end
+  return nil if matrix.empty?
+
+  rows = matrix.size
+  columns = matrix[0].size
+
+  rows.times do |row|
+    columns.times do |column|
+      matrix[row][column] = "ZERO" if matrix[row][column] == 0
+    end
+  end
+
+  rows.times do |row|
+    columns.times do |column|
+      next unless matrix[row][column] == "ZERO"
+
+      columns.times { |i| matrix[row][i] = 0 unless matrix[row][i] == "ZERO" }
```

Comment:
Personal preference nit: some programmers (e.g. many Java people) might find this a bit _too_ concise, and will prefer something more explicit.

Rubyists, unlike Java people, sometimes _embrace_ this conciseness even if it makes code take slightly more time to understand - as it results in less code needing to be understood. 🙂

(Of course - like all personal preferences, this one's totally up to you.)

Reply:
Thank you! It's really helpful to hear about style preferences. When I work on this sort of thing in a vacuum, I can get totally drawn into the temptation to make it beauuuuuutifully elegant and forget about clarity/other people 😮 🙂

Reply:
Yep - and style preferences depend heavily on the existing codebase. Many codebases/companies will have style guides, but inevitably your first few PRs at a new company  (or even a new codebase within the same company! e.g. iOS vs. Android) will be spent learning (and internalizing) that codebases' specific guidelines/patterns.